File: src/main/java/com/smarthome/backend/model/SystemSettings.java
package com.smarthome.backend.model;

import lombok.Data;

/**
 * System-Einstellungen, einschließlich Versionsinformationen und Auto-Update-Einstellungen.
 */
@Data
public class SystemSettings {
    private VersionInfo frontend;
    private VersionInfo backend;
    private Boolean autoupdate;
    private UpdateTimes updatetimes;
    private String serverIp;
}

File: src/main/node/com/smarthome/backend/server/api/entities/settings/SystemSettings.ts
import { VersionInfo } from "./VersionInfo.js";
import { UpdateTimes } from "./UpdateTimes.js";

export class SystemSettings {
  frontend?: VersionInfo;
  backend?: VersionInfo;
  autoupdate?: boolean;
  updatetimes?: UpdateTimes;
  serverIp?: string;

  constructor(init?: Partial<SystemSettings>) {
    Object.assign(this, init);
  }
}
